src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Usage/PreferGenericOverloadsTests.cs

```diff
@@ -0,0 +1,2409 @@
+                class A {}
+                class B {}
+
+                class C
+                {
+                    void M(System.Type type1, System.Type type2, System.Type type3, int x) {}
+                    void M<T1, T2, T3>(int x) {}
+
+                    void Test()
+                    {
+                        M<A, B, C>(0);
+                    }
+                }
+                """;
+
+            await VerifyCS.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task SingleTypeArgumentWithOtherArguments_OffersFixer_CS()
+        {
+            string source = """
+                class C
+                {
+                    void M(System.Type type, int x, string y, object z) {}
+                    void M<T>(int x, string y, object z) {}
+
+                    void Test()
+                    {
+                        [|M(typeof(C), 0, "Test", this)|];
+                    }
+                }
+                """;
+
+            string fixedSource = """
+                class C
+                {
+                    void M(System.Type type, int x, string y, object z) {}
+                    void M<T>(int x, string y, object z) {}
+
+                    void Test()
+                    {
+                        M<C>(0, "Test", this);
+                    }
+                }
+                """;
+
+            await VerifyCS.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task MultipleTypeArgumentsWithOtherArguments_OffersFixer_CS()
+        {
+            string source = """
+                class A {}
+                class B {}
+
+                class C
+                {
+                    void M(System.Type type1, System.Type type2, System.Type type3, int x, string y, object z) {}
+                    void M<T1, T2, T3>(int x, string y, object z) {}
+
+                    void Test()
+                    {
+                        [|M(typeof(A), typeof(B), typeof(C), 0, "Test", this)|];
+                    }
+                }
+                """;
+
+            string fixedSource = """
+                class A {}
+                class B {}
+
+                class C
+                {
+                    void M(System.Type type1, System.Type type2, System.Type type3, int x, string y, object z) {}
+                    void M<T1, T2, T3>(int x, string y, object z) {}
+
+                    void Test()
+                    {
+                        M<A, B, C>(0, "Test", this);
+                    }
+                }
+                """;
+
+            await VerifyCS.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task SingleTypeArgumentWithOtherGenericArgument_OffersFixer_CS()
+        {
+            string source = """
+                class C
+                {
+                    void M(System.Type type, object x) {}
+                    void M<T>(T x) {}
```

Comment:
Wonder if this scenario cause more false positives ...

Reply:
IIRC this test was motivated by the following (from roslyn-cs.txt):
W:\roslyn\src\Compilers\CSharp\Test\Semantic\SourceGeneration\GeneratorDriverTests.cs(3102,78): info CA2262: Prefer the generic overload 'System.Enum.GetName<TEnum>(TEnum)' instead of 'System.Enum.GetName(System.Type, object)'

Reply:
`System.Enum.GetName<TEnum>(TEnum)` instead of `System.Enum.GetName(System.Type, object)` is a good example, thank you!